### DISCUSSION TITLE
Can I send Multiple Messages for the same destination in Mirth

### DISCUSSION THREAD

**Discussion by IbrahimAlhaj9** [nextgenhealthcare/connect#5175] [General]
Hi Everyone,
I have a unique input message and I want to translate it into multiple messages for the same destination

Input message:
```
MSH|^~\&|||Novo|NextGen|20201102154519|||PS20-00065201102154520317|P|||\\osd7944\Novoh\InfDXHL7\HL7Results\D365\PDF\PS20-00065.pdf
OBR|1|778798798798|PS20-00065^NovoPath|^^Pathology Report|||20200930000000|20200930123500||||||20200930122700||1134314263^Moore^Mary^^^MMoore1^^N^^|||Bonemarrow|F||20201102153800|||C|Signed off||1992116750^Wilking^Ashley^^^AWilking^^N^^||||^||
OBX|1|TX|990100^History1|||||A|||C|||20200930000000
OBX|1|TX|990100^History2|||||A|||C|||20200930000000
OBX|1|TX|990100^History3|||||A|||C|||20200930000000

```

Output messages files :
```
MSH|^~\&|||Novo|NextGen|20201102154519|||PS20-00065201102154520317|P|||\\osd7944\Novoh\InfDXHL7\HL7Results\D365\PDF\PS20-00065.pdf
OBR|1|778798798798|PS20-00065^NovoPath|^^Pathology Report|||20200930000000|20200930123500||||||20200930122700||1134314263^Moore^Mary^^^MMoore1^^N^^|||Bonemarrow|F||20201102153800|||C|Signed off||1992116750^Wilking^Ashley^^^AWilking^^N^^||||^||
OBX|1|TX|990100^History1|||||A|||C|||20200930000000

```
```
MSH|^~\&|||Novo|NextGen|20201102154519|||PS20-00065201102154520317|P|||\\osd7944\Novoh\InfDXHL7\HL7Results\D365\PDF\PS20-00065.pdf
OBR|1|778798798798|PS20-00065^NovoPath|^^Pathology Report|||20200930000000|20200930123500||||||20200930122700||1134314263^Moore^Mary^^^MMoore1^^N^^|||Bonemarrow|F||20201102153800|||C|Signed off||1992116750^Wilking^Ashley^^^AWilking^^N^^||||^||
OBX|1|TX|990100^History2|||||A|||C|||20200930000000
```
```
MSH|^~\&|||Novo|NextGen|20201102154519|||PS20-00065201102154520317|P|||\\osd7944\Novoh\InfDXHL7\HL7Results\D365\PDF\PS20-00065.pdf
OBR|1|778798798798|PS20-00065^NovoPath|^^Pathology Report|||20200930000000|20200930123500||||||20200930122700||1134314263^Moore^Mary^^^MMoore1^^N^^|||Bonemarrow|F||20201102153800|||C|Signed off||1992116750^Wilking^Ashley^^^AWilking^^N^^||||^||
OBX|1|TX|990100^History3|||||A|||C|||20200930000000
```

Mirth  source type: HTTP Listener
Mirth Destination type:JavaScript Writer.

Please help me
Thank you In advance.

**Comment by jonbartels** [comment]
I have some ideas but not a complete solution:
* A custom batch splitter might work. Logically, that is what you are doing. 
* Receive the message and have a JS Writer destination that loops over OBX segments. For each OBX segment construct a new HL7 message then call `router.routeMessage` to send that smaller HL7 message to a different channel.

**Reply by IbrahimAlhaj9** [reply]
@jonbartels thanks i will try try solution

**Comment by tonygermano** [comment]
I'm very confused about what your request is. Are you asking how to create the individual messages from one or how to send them after they are split?

**Reply by IbrahimAlhaj9** [reply]
@tonygermano , i need to split the message into multiple message then send them into multiple files for the same destination
Thank you

**Comment by IbrahimAlhaj9** [comment]
@jonbartels , @tonygermano 
i have the following code into the javascript writer:

```
var FileName = $('FileName');

var encData = FileUtil.encode(new java.lang.String(SerializerFactory.getSerializer('HL7V2').fromXML(connectorMessage.getEncodedData())).getBytes())

var ret = new String('<Message>')
var datestamp=DateUtil.getCurrentDate('yyyyMMddHHmmssSSS') 

ret+='<Body FileName="'+ FileName + '.pdf' + '" Type="hl7" Include="true">' +  encData + '</Body>'


ret+='<Files>'


ret+='</Files>'

ret+='</Message>'

$c('chanretChannelMessage','Success')
$c('chanretLastChannelStep','Destination Writer')
$c('chanretStatusCode','200')
return XmlUtil.prettyPrint(ret)
```

this code is work properly when I send the translate message to my destination
but how I can send multiple messages for the same destination using the same code?

@jonbartels ```router.routeMessage``` allow us to select an another channel, I need to use the same channel

Thank you

**Comment by odoodo** [comment]
Using 2 channels would indeed be the best solution. If this really isn't possible:
Assemble your messages in a transformer, write the results to e.g. an ArrayList and place the ArrayList in the channel map.
In the javascript destination read the messages from the channel map and do the intended sending for each message.
